scaleway/resource_k8s_cluster_beta.go

```diff
@@ -281,6 +339,25 @@ func resourceScalewayK8SClusterBetaCreate(d *schema.ResourceData, m interface{})
 
 	req.AutoscalerConfig = autoscalerReq
 
+	autoUpgradeEnable, okAutoUpgradeEnable := d.GetOk("auto_upgrade.0.enable")
+	autoUpgradeStartHour, okAutoUpgradeStartHour := d.GetOk("auto_upgrade.0.maintenance_window_start_hour")
+	autoUpgradeDay, okAutoUpgradeDay := d.GetOk("auto_upgrade.0.maintenance_window_day")
+
+	// check if all or 0 values are set
+	if (okAutoUpgradeDay || okAutoUpgradeEnable || okAutoUpgradeStartHour == true) && (okAutoUpgradeDay && okAutoUpgradeEnable && okAutoUpgradeStartHour == false) {
```

Comment:
nit: remove ` == true`

Reply:
something is wrong with this condition `(a or b or c) and (a and b and !c)` is the same as `a and b and !c`

Reply:
reminder: https://www.tutorialspoint.com/go/go_operators_precedence.htm